activate physical card

MrMuzyk opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`


Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/22878
PROPOSAL: 


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
4. Verify a modal appears displaying a preview of that image
--->

- [x] Add useEffect with test data to the `InitialSettingsPage`
```
    useEffect(() => {
        window.Onyx.merge(`cardList`, {
            1234: {
                cardID: 1234,
                state: 4,
                bank: '1000',
                availableSpend: 10000,
                domainName: 'Expensify',
                lastFourPAN: '',
                isVirtual: false,
            },
        });

        Navigation.navigate(ROUTES.SETTINGS_WALLET_DOMAINCARDS.getRoute('expensify'));
    }, []);
```
- [x] Open app
- [x] Press on your avatar to open Settings
- [x] Press activate physical card button
- [x] Enter random 4 digit code
- [x] Error message about incorrect code should appear

In order to test happy path
- [x] Change failureData inside `Card.js` to
```
            failureData: [
                {
                    onyxMethod: Onyx.METHOD.MERGE,
                    key: ONYXKEYS.CARD_LIST,
                    value: {
                        [cardID]: {
                            isLoading: false,
                            cardID: 1234,
                            state: 3,
                            bank: '1000',
                            availableSpend: 10000,
                            domainName: 'Expensify',
                            lastFourPAN: '',
                            isVirtual: false,
                        },
                    },
                },
            ],
```
- [x] Relog
- [x] Press on your avatar to open Settings
- [x] Press activate physical card button
- [x] Enter random 4 digit code
- [x] User should get redirected back to Expensify card page and `Activate physical card` button shouldn't be visible
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/99fff72f-218c-4f96-8f36-808fa0a9385d

Error path
https://github.com/Expensify/App/assets/23487612/f043c39c-18dc-41c3-983c-dcc6533f4002


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/2c00585e-3e24-4f96-b3e6-ab7f0cfa2f7e

Error path
https://github.com/Expensify/App/assets/23487612/033e7634-0bb2-4b4f-aa4f-00474ea2e640

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/3218c1fc-fce1-48c1-b36b-c69b12aabad3

Error path
https://github.com/Expensify/App/assets/23487612/4037e721-0440-462f-aa1f-d049467b0eaa

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/0fc7d45a-814f-44a3-b234-1a30e3dbc278

Error path
https://github.com/Expensify/App/assets/23487612/aa59912b-ac69-4537-9706-d7075c3cf362

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/ce997f17-f882-4b0e-880d-1f2bb1a55bea

Error path
https://github.com/Expensify/App/assets/23487612/dc581a4c-086f-4ba7-bc64-2fcdc069caed

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->
Happy path
https://github.com/Expensify/App/assets/23487612/6761aca1-79a0-4f68-bb54-4ca4b099e239

Error path
https://github.com/Expensify/App/assets/23487612/437debbd-7b47-4691-902c-ad0431b6f8e9

</details>

MrMuzyk committed:
feat 22878 activate physical card flow

MrMuzyk committed:
Merge remote-tracking branch 'eugenefork/feature/expensify-card' into feat-22878-activate-physical-card

MrMuzyk committed:
feat: stash new changes

MrMuzyk committed:
Merge branch 'feature/expensify-card' of https://github.com/pasyukevich/Expensify-app into feat-22878-activate-physical-card

MrMuzyk committed:
feat: remove commented out line

MrMuzyk committed:
Merge branch 'feature/expensify-card' of https://github.com/pasyukevich/Expensify-app into feat-22878-activate-physical-card

MrMuzyk committed:
feat: rebased with card page branch

MrMuzyk committed:
feat: update es translations

MrMuzyk committed:
fix: remove test data

MrMuzyk committed:
fix: cr fixes

MrMuzyk committed:
fix: typefix

MrMuzyk committed:
fix: another batch of cr fixes

MrMuzyk committed:
fix: resolved native focus edge case

MrMuzyk committed:
fix: renamed one const variable

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk commented:
I have read the CLA Document and I hereby sign the CLA

MrMuzyk committed:
fix: focus edge case fix

fedirjh reviewed (commented):
Left some comments. I am also seeing this console error : 

<img width="623" alt="Screenshot 2023-09-28 at 12 00 07 PM" src="https://github.com/Expensify/App/assets/36869046/e505fce1-506f-452e-bf58-c65acc4a3a15">

MrMuzyk commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,177 @@
+    cardList: {},
+};
+
+const LAST_FOUR_DIGITS_LENGTH = 4;
```
Is there a need for that? It is only being used in this file and not anywhere else.

MrMuzyk commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,177 @@
+
+function ActivateCardPage({
+    cardList,
+    route: {
+        params: {domain},
+    },
```
This is a required prop. I dont think there is a need for default value here

MrMuzyk commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,177 @@
+    /**
+     * Set form error if there is one for cardID in onyx
+     */
+    useEffect(() => {
+        if (isUndefined(lodashGet(cardList[cardID], 'errors', undefined))) {
+            setFormError('');
+        } else {
+            setFormError(ErrorUtils.getLatestErrorMessage(cardList[cardID]));
+        }
+    }, [cardList, cardID]);
```
Would you mind explaining what is your idea here?

MrMuzyk replied:
We need to clear the error on any input after it is shown so state seems like a sensible solution here

MrMuzyk commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,177 @@
+        setLastFourDigits(text);
+    };
+
+    const submitAndNavigateToNextPage = useCallback(() => {
+        activateCardCodeInputRef.current.blur();
```
ok, should we disable the submit button untill 4 digits are entered or we want to have an early validation before making API call here in submit method?

MrMuzyk replied:
Whats the offline flow here? It wasnt mentioned at all in the docs

MrMuzyk commented on src/pages/settings/Wallet/ActivateCardPage.js:
```diff
@@ -0,0 +1,177 @@
+        <IllustratedHeaderPageLayout
+            title={translate('activateCardPage.activateCard')}
+            onBackButtonPress={() => {
+                CardSettings.clearCardListErrors(cardID);
```
is user tries to submit wrong code, then goes back and then comes to this page again he's greeted by last error as its still stored in onyx

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk commented:
> Left some comments. I am also seeing this console error :
> 
> <img alt="Screenshot 2023-09-28 at 12 00 07 PM" width="623" src="https://user-images.githubusercontent.com/36869046/271266351-e505fce1-506f-452e-bf58-c65acc4a3a15.png">

this was caused by something in main. Ive rebased locally with the lastest one and now the error is gone. I'll make sure to check again once I fix everything pointed out during code review

MrMuzyk committed:
fix: code review fixes

MrMuzyk committed:
fix: early return for card state check hook

fedirjh reviewed (commented):
We have a bug with numberpad on the mobile web (safari and Chrome) , the number button doesn’t work after the input loses focus 

https://github.com/Expensify/App/assets/36869046/7de4edfb-51bd-4a85-a3fe-5b5cd70c70a1

---

On IOS, the submit button is not docked at the bottom :

| List page | validate page |
|--------|--------|
| <img width="428" alt="Screenshot 2023-09-29 at 12 51 59 PM" src="https://github.com/Expensify/App/assets/36869046/6af5d3de-b015-4299-82c4-a816f1957a4b"> | <img width="472" alt="Screenshot 2023-09-29 at 12 52 04 PM" src="https://github.com/Expensify/App/assets/36869046/5376df6c-181f-43bb-8a70-2b309e2070c3"> | 

---

On IOS Safari, the validate button is initially hidden, this may not be a bug , seems like a limitation with the design, 

https://github.com/Expensify/App/assets/36869046/562c21bc-9b26-4ddd-81e6-b1ceaabb0f54

---

On mobile, when an error occurs the design expands, is that expected? cc @shawnborton 


https://github.com/Expensify/App/assets/36869046/e702ca15-1afc-45d4-a793-2be576d45fb9

MrMuzyk commented on src/pages/settings/Wallet/ActivatePhysicalCardPage.js:
```diff
@@ -0,0 +1,175 @@
+    /**
+     * Set form error if there is one for cardID in onyx
+     */
+    useEffect(() => {
+        if (isUndefined(lodashGet(cardList[cardID], 'errors', undefined))) {
+            setFormError('');
+        } else {
+            setFormError(ErrorUtils.getLatestErrorMessage(cardList[cardID]));
+        }
+    }, [cardList, cardID]);
```
I'll look into it and try to change it around

MrMuzyk replied:
But yeah, looking at the comment few lines below. Splitting the onyx error and local error might be the way to go here

MrMuzyk commented on src/languages/en.ts:
```diff
@@ -817,6 +817,14 @@ export default {
+        pleaseEnterLastFour: 'Please enter the last four digits of your card.',
+        activatePhysicalCard: 'Activate physical card',
+        error: {
+            notEnoughDigits: 'Not enough digits.',
```
Yes, will do. 

The specs for this issue state:

>The errors returned here are basically just “not enough digits” (must be 4) and “incorrect last four digits”. 

Thought that it's fine for me to put it in there without asking any1 for confirmation.

MrMuzyk replied:
https://github.com/Expensify/App/issues/22878#issuecomment-1740842419

MrMuzyk commented:
@fedirjh 

>On IOS, the submit button is not docked at the bottom :

Should it also be docked at the bottom on web?

<img width="378" alt="image" src="https://github.com/Expensify/App/assets/23487612/094690de-34e4-4974-b67c-3892847f8e32">

MrMuzyk committed:
fix: fixed focus issues on mobile safari/chrome, fixed layout

MrMuzyk commented:
@fedirjh 

I've resolved focus and layout issues. 'Activate physical card' is now docked at the bottom and layout is not moving when error is appearing/disappearing so its an improvement compared to previous version.

Apart from that Ive split the error messages like you've suggested

Focus issue resolved

https://github.com/Expensify/App/assets/23487612/fdd07853-4820-4d5f-bfd2-21de09f8654b

https://github.com/Expensify/App/assets/23487612/36dd4b70-4cad-4d01-8f2f-74e433c2dcdc

Layout issue resolved

https://github.com/Expensify/App/assets/23487612/68f106bf-171c-438b-a706-92e30673d2f7

https://github.com/Expensify/App/assets/23487612/f8141bba-59eb-4052-89a3-698663c9c4d7

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk committed:
fix: old focus edge case solution and new error translation

MrMuzyk committed:
feat: updated translation

fedirjh reviewed (commented):
Looks good.

MrMuzyk commented on src/pages/settings/Wallet/ActivatePhysicalCardPage/index.js:
```diff
@@ -0,0 +1,182 @@
+     *
+     * @param {String} text
+     */
+    const onCodeInput = (text) => {
```
Sorry, Ive missed that. I was following the details from page 50 where this feature is described in detail and it wasnt mentioned there. Ill fix that

MrMuzyk committed:
fix: move styles helper function and supply onfulfill prop to magic input

MrMuzyk commented:
@fedirjh I've moved helper to src/styles and added prop for MagicInput component that submits its value automatically once all digits are provided

MrMuzyk commented:
https://github.com/Expensify/App/assets/23487612/b8becf4c-cb59-4f93-a383-c20fd26ff875

shawnborton commented:
Oh interesting, in that case, I think we should make sure the green button just sits below the number pad and is not docked to the bottom of the view. Thoughts? cc @Expensify/design

dubielzyk-expensify commented:
Agree with you, Shawn. Prioritize the input over the button

MrMuzyk commented:
>Oh interesting, in that case, I think we should make sure the green button just sits below the number pad and is not docked to the bottom of the view. Thoughts? cc @Expensify/design
And that would be approach for all platforms?

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk committed:
feat: update layout

MrMuzyk commented:
https://github.com/Expensify/App/assets/23487612/c6aeb6dd-97cf-4dc2-b28e-a2b7664cd46a


https://github.com/Expensify/App/assets/23487612/143cad1d-f6a0-4a27-a8b7-0fc44587a603


https://github.com/Expensify/App/assets/23487612/6a04489b-136b-4dd8-98b5-4668136ecb97


https://github.com/Expensify/App/assets/23487612/be655899-6be9-47d5-9f94-2c66e82f2bae


https://github.com/Expensify/App/assets/23487612/b01cfae1-3341-4023-a454-5eb1c5b28d93


https://github.com/Expensify/App/assets/23487612/7441638f-d339-4a11-883f-e661f33e14d2

@fedirjh I've updated the layout according to the latest guidelines. Submit button should be only docked at the bottom if there is an empty space above it. I've uploaded the videos to showcase how it looks like now. PR is ready for the final review

MrMuzyk commented:
@fedirjh 
Is there anything else I can do apart from this check which is already in codebase?

```    
if (_.isEmpty(physicalCard)) {
        return <NotFoundPage />;
}

```

This page will not work properly without a physicalCard object provided from onyx

shawnborton commented:
Looking good! Minor feedback, but can we reduce the gaps between the activation number inputs to be just 12px like our mockup? Right now the gaps are a bit wide:
<img width="490" alt="CleanShot 2023-10-06 at 07 50 48@2x" src="https://github.com/Expensify/App/assets/2319350/5da77087-4131-42dc-9624-334061d02754">

fedirjh reviewed (commented):
> Is there anything else I can do apart from this check which is already in codebase?

Apparently, using `cardList[cardID]` is causing this crash, I see using `physicalCard` should fix the crash.

fedirjh reviewed (commented):
Looking good.

MrMuzyk committed:
fix: bigger inputs, dry code and undefined cardID

MrMuzyk commented:
@shawnborton 

Updated the inputs

<img width="379" alt="Screenshot 2023-10-06 at 15 39 15 1" src="https://github.com/Expensify/App/assets/23487612/1e711641-0e2c-4a2c-89e1-25a2631a3512">
<img width="476" alt="Screenshot 2023-10-06 at 15 39 42" src="https://github.com/Expensify/App/assets/23487612/7408bdc7-868c-4ee8-891f-fb22f7d0550e">

@fedirjh 
Applied requested changes. Thanks for all the tips and making me realise of all the different scenarios that had to be checked here. Haven't thought about deep linking to that page .

shawnborton commented:
Awesome, thanks!

fedirjh reviewed (approved):
The code looks good to me and tests well.

grgia commented on src/CONST.ts:
```diff
@@ -83,6 +83,14 @@ const CONST = {
 
     MERCHANT_NAME_MAX_LENGTH: 255,
 
+    CARD_STATE: {
+        OPEN: 3,
+        NOT_ACTIVATED: 4,
+        DEACTIVATED: 5,
+        CLOSED: 6,
+        SUSPENDED: 7,
+    },
```
Could you please use `CONST.EXPENSIFY_CARD.STATE` instead

MrMuzyk replied:
yes ofc. It wasn't in main at the time of creating this

MrMuzyk committed:
Merge branch 'main' of github.com:MrMuzyk/App into feat-22878-activate-physical-card

MrMuzyk committed:
fix: removed card state const and swap to new added const, reverted change in onyxkeys for cardlist

MrMuzyk commented:
@grgia removed new const I've added and swapped to the one you've proposed. On top of that I've spotted that the type in ONYXKEYS was unnecessarily changed by me so I've reverted that. Rebased with latest main as well :)

grgia reviewed (approved):
Great work on this!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6477981814) to staging by https://github.com/grgia in version: 1.3.81-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6491054357) to staging by https://github.com/grgia in version: 1.3.83-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
